Libraries/LibWebView/ViewImplementation.cpp

```diff
@@ -667,6 +667,24 @@ void ViewImplementation::do_not_track_changed()
     client().async_set_enable_do_not_track(page_id(), do_not_track == DoNotTrack::Yes);
 }
 
+void ViewImplementation::dns_settings_changed()
```

Comment:
hmm given that this doesn't touch the WebContent process, I think it wants a different home. Because if we have 10 tabs open when DNS settings change, this will run 10 times. You could make `WebView::Application` itself a settings observer to get this notification.

Reply:
Making Application a SettingsObserver introduces a dependency cycle (Application ctor calls SettingsObserver ctor, which expects the Application ctor to have been run at least up to setting `s_the`)
I added a workaround, but not exactly the cleanest thing around